t/plugin/openid-connect.t

```diff
@@ -626,6 +628,86 @@ Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhMSI6IkRhdGEgM
 
 
 
+=== TEST 9z: update plugin public key, so tokens can be validated locally
```

Comment:
Why using `9z`?

Reply:
I'm in the middle of the process of adding cases, so will fix the numbering when done. No point doing it now as I don't know how many cases I will add.

Also, why number the cases in the first place? If you insert just one somewhere you have to adjust the numbers of all cases below.

Reply:
Also, if anyone could help me setting up a proper dev environment so I don't have to go through the entire suite of GitHub checks all the time just to run the tests on the oidc module, I would appreciate that very much.

I tried setting up an Ubuntu 18.04 VM according to Travis configuration, but it's frankly a nightmare to get everything up and running because most scripts need tiny little adjustments to actually work as expected in that environment. In the end, I seemingly managed to get all containers and everything up and running, but the tests still wouldn't run because the APISIX process was still throwing up a lot in the error log. Not a single test passed as a result.

I need to find a better development workflow or I may have to abandon this whole ticket. At this stage, I'm wasting too much time on stuff not really related to the actual dev work. Can't afford that any longer.

Maybe it might be a good idea to set up a Vagrant box or similar so new developers don't have to go through all this trial and error.

Any hints appreciated.

Reply:
It is amazing that you finally get all the test passed! Thanks a lot!

Reply:
@jenskeiner 
We can fulfill the tests for you later. Before we can merge, the only thing you need to do is to update the English doc: https://github.com/apache/apisix/blob/master/doc/plugins/openid-connect.md#attributes

Reply:
Hang on, I disabled most of the tests, so currently just running the OIDC ones. Need to revert that before we proceed.

Also, was thinking how to test the path where the initial request doesn't have a token and is redirected to the identity provider. When the user has authenticated, the identity provider will call the redirect URL with a code so that APISIX (the relying party) can obtain access token, ID token, and userinfo from the provider's endpoints.

We need to test this because the plugin wil set headers to the upstream service accordingly in this case.

Probably best to mock up the endpoints instead of interacting with the Keycloak container. I'll see if I can get this done as well.

In any case, I will post here when I believe the PR is actually ready for final review round.